src/modules/ekf2/EKF2.cpp

```diff
@@ -1123,9 +1082,9 @@ void EKF2::Run()
 				test_ratios.fake_hvel[0] = test_ratios.fake_hvel[1] = test_ratios.fake_vvel = NAN;
 
 				// calculate noise filtered velocity innovations which are used for pre-flight checking
-				if (_vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_STANDBY) {
+				if (_standby) {
 					float dt_seconds = imu_sample_new.delta_ang_dt;
-					runPreFlightChecks(dt_seconds, control_status, _vehicle_status, innovations);
+					runPreFlightChecks(dt_seconds, control_status, innovations, _can_observe_heading_in_flight);
```

Comment:
@bresch something minor I noticed is that `vehicle_local_position` is being published containing `_preflt_checker` results before the preflight checks run (always 1 update behind).